src/integrationTest/groovy/UmplePluginIntegrationTests.groovy

```diff
@@ -0,0 +1,88 @@
+import org.junit.*
+import static org.junit.Assert.*
+import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.api.Project
+import org.gradle.tooling.* //GradleConnector, ProjectConnection, BuildLauncher
+
+public class UmpleIntegrationTests {
+    private static Project testProject
+    private static final DIRECTORIES_TO_CHECK = 
+    [
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}subproj${File.separator}build${File.separator}classes${File.separator}generatedSource",
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}subproj${File.separator}generated${File.separator}java",
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}sub2${File.separator}build${File.separator}classes${File.separator}generatedSource",
+        "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject${File.separator}sub2${File.separator}generated${File.separator}java"
+    ] 
+    private static final PATH_TO_TEST_PROJECT = "src${File.separator}integrationTest${File.separator}resources${File.separator}TestProject"
+    private static final TEST_PROJECT_TASK = "compileJava"
+
+    @Before
+    public void setUp() {
+        testProject = ProjectBuilder.builder().build()
+        testProject.plugins.apply 'umple.gradle.plugin'
+    }
+
+    @Test
+    public void applyUmplePluginToProject() {      
+        assertTrue('Umple plugin was not applied', testProject.getPlugins().hasPlugin('umple.gradle.plugin'))
```

Comment:
Yup! All you'd need to do to make the entire `UmplePluginIntegrationTests` file work in Java is add `;`s and change the way the `DIRECTORIES_TO_CHECK` list is initialized. `assertTrue` is a JUnit function.